### PR TITLE
 Migrating Task-Status-Selector From AngularJS to Angular

### PR DESCRIPTION
**Pull request by BaderAlotaibi**
# Description

Converting The HTMl AngularJs to HTML Angular
and Converting the Coffee script to Typescript 
And do the needed changes to the related files.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X ] Bug fix (non-breaking change which fixes an issue)
 
# How Has This Been Tested?
run the server and there was no issue,
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

If you have any questions, please contact @macite or @jakerenzella.

### CODE REVIEW

**Comment by BaderOtaibi** [comment]
Hi Dr Jake 
Please accept my apologies, Mistakenly I want to comit the message as Task-Status-Selector
But I put 
Task-Status-Card 

So please consider the commit message for a Task-Status-Selector
As I am not able to do any change because I have submitted my pull request to the panel and also in my individual task. 
The problem is only in the message in the commit not in the code .

**Comment by jakerenzella** [comment]
We don't want misleading commits in the history, please resubmit this PR with the correct commit name

File: src/app/tasks/task-status-selector/task-status-selector.component.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <li class="col-xs-6 col-sm-4 col-md-4" *ngFor="let action of task.filterFutureStates(studentTriggers)" > <!--taskEngagementConfig.studentTriggers-->
+      <div class="row">
+        <button class="col-xs-12 btn task-status {{action.class}}" [ngClass]="{'active' : project.selectedTask.status == action.status}" (click)="triggerTransition(action.status)" [(ngModel)]="project.selectedTask.status" btn-radio="action.status" >
+          <i class="{{action.icon}} fa-2x task-status-icon"></i>
+          <strong>{{action.label}}</strong>
+        </button>
+      </div>
+    </li><!--/student-statuses-->
+  </ul>
+
```
Comment (jakerenzella) [inline src/app/tasks/task-status-selector/task-status-selector.component.html:30]: This still uses bootstrap styles and old icons, should be angular material mat-icon and styles

**Review by jakerenzella — CHANGES_REQUESTED** [review]

**Comment by BaderAlotaibi** [comment]
> We don't want misleading commits in the history, please resubmit this PR with the correct commit name

For This request , It is Done 
for Converting the bootstrap , I will work on it now , and sorry for this as I have not thought about it

**Comment by BaderAlotaibi** [comment]
image
This is a result of compiling the code after making the changes

**Comment by jakerenzella** [comment]
Not up to acceptable standards, misleading commit history and new warnings introduced.

**Pull request closed without merging**